app/Http/Controllers/API/v1/StatisticsController.php

```diff
@@ -36,8 +39,62 @@ public function leaderboardFriends(): AnonymousResourceCollection {
     /**
      * @return AnonymousResourceCollection
      */
-    public function leaderboardForMonth(string $date) {
+    public function leaderboardForMonth(string $date): AnonymousResourceCollection {
         $date = Carbon::parse($date);
         return LeaderboardUserResource::collection(LeaderboardBackend::getMonthlyLeaderboard(date: $date));
     }
+
+    /**
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function getPersonalStatistics(Request $request): JsonResponse {
+        $validated = $request->validate([
+                                            'from'  => ['nullable', 'date'],
+                                            'until' => ['nullable', 'date', 'after_or_equal:from']
+                                        ]);
+
+        $from  = isset($validated['from']) ? Carbon::parse($validated['from']) : Carbon::now()->subWeeks(4);
+        $until = isset($validated['until']) ? Carbon::parse($validated['until']) : Carbon::now();
+
+        $purposes   = StatisticsTravelPurposeResource::collection(
+            StatisticBackend::getTravelPurposes(user: auth()->user(), from: $from, until: $until)
+        );
+        $categories = StatisticBackend::getTopTravelCategoryByUser(user: auth()->user(), from: $from, until: $until);
+        $operators  = StatisticBackend::getTopTripOperatorByUser(user: auth()->user(), from: $from, until: $until);
+        $travelTime = StatisticBackend::getDailyTravelTimeByUser(user: auth()->user(), from: $from, until: $until);
+
+        $returnData = [
+            'purpose'    => $purposes,
+            'categories' => $categories,
+            'operators'  => $operators,
+            'time'       => $travelTime
+        ];
+
+        $additionalData = [
+            'meta' => [
+                'from'  => $from,
+                'until' => $until
+            ]
+        ];
+
+        return $this->sendv1Response(data: $returnData, code: 200, additional: $additionalData);
+
+    }
+
+    public function getGlobalStatistics(): JsonResponse {
+        $from  = Carbon::now()->subWeeks(4);
+        $until = Carbon::now();
```

Comment:
why static?

Reply:
Because I don't want to create a possibility of overloading the server when users select a date range